### PR TITLE
[backport ipa-4-7] one-way trust to AD with shared secret

### PR DESCRIPTION
**Pull request by abbra**
Backport of https://github.com/freeipa/freeipa/pull/2926 to ipa-4-7

### CODE REVIEW

**Comment by abbra** [comment]
@tiran, Python 2 linter in Travis CI produces an error that I cannot reproduce with F29 python2-lint myself.

File: ipaserver/install/plugins/adtrust.py [hunk 1]
```diff
@@ -1,30 +1,26 @@
...
-# (at your option) any later version.
-#
-# This program is distributed in the hope that it will be useful,
-# but WITHOUT ANY WARRANTY; without even the implied warranty of
-# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-# GNU General Public License for more details.
-#
-# You should have received a copy of the GNU General Public License
-# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+# Copyright (C) 2012-2019  FreeIPA Contributors see COPYING for license
```
Comment (tiran) [inline ipaserver/install/plugins/adtrust.py]: Add ``from __future__ import absolute_import`` after the header to fix the pylint issue.

**Comment by tiran** [comment]
I was able to push ``from __future__ import absolute_import`` into your branch.

**Comment by tiran** [comment]
ipa-4-7:

* 50c0d6dc7b93abe2448ecc499e32e11bf5e3e7f5 Add design page for one-way trust to AD with shared secret
* f6183c80826c90a8b0c051284d66cb0a2378cdc1 trust: allow trust agents to read POSIX identities of trust
* 0b5ae1bd00c336c1963efdb28891780fb61af17e trusts: add support for one-way shared secret trust
* 2a7731c5ae8d3f7fad9ee20584e271100d090557 upgrade: upgrade existing trust agreements to new layout
* 595f42af25f882fe5d45b0e08bdd5300d7267fdb upgrade: add trust upgrade to actual upgrade code

**Pull request closed without merging**